src/LuaPrinter.ts

```diff
@@ -436,7 +468,7 @@ export class LuaPrinter {
         }
     }
 
-    private printFunctionExpression(expression: tstl.FunctionExpression): SourceNode {
+    private printFunctionExpression(expression: tstl.FunctionExpression, name?: SourceChunk): SourceNode {
```

Comment:
I think for plugin/overriding printer reasons it is best not to pollute our interface with random function arguments (similar to what we discussed recently in the transformer visitor pattern), surely there is a better way to do this right?

Reply:
I'll move this into it's own `printFunctionDeclaration` method